zerver/lib/actions.py

```diff
@@ -1926,18 +1926,34 @@ def do_send_messages(
         """
         user_ids = send_request.active_user_ids | set(user_flags.keys())
 
+        # Sender at the start of the users list so that he gets the
+        # message first and perseived latency will be less
+        sender_id = send_request.message.sender.id
+
         users = [
             dict(
-                id=user_id,
-                flags=user_flags.get(user_id, []),
-                always_push_notify=(user_id in send_request.push_notify_user_ids),
-                stream_push_notify=(user_id in send_request.stream_push_user_ids),
-                stream_email_notify=(user_id in send_request.stream_email_user_ids),
-                wildcard_mention_notify=(user_id in send_request.wildcard_mention_user_ids),
+                id=sender_id,
+                flags=user_flags.get(sender_id, []),
+                always_push_notify=(sender_id in send_request.push_notify_user_ids),
+                stream_push_notify=(sender_id in send_request.stream_push_user_ids),
+                stream_email_notify=(sender_id in send_request.stream_email_user_ids),
+                wildcard_mention_notify=(sender_id in send_request.wildcard_mention_user_ids),
             )
-            for user_id in user_ids
         ]
 
+        for user_id in user_ids:
+            if user_id != sender_id:
+                users.append(
+                    dict(
+                        id=user_id,
+                        flags=user_flags.get(user_id, []),
+                        always_push_notify=(user_id in send_request.push_notify_user_ids),
+                        stream_push_notify=(user_id in send_request.stream_push_user_ids),
+                        stream_email_notify=(user_id in send_request.stream_email_user_ids),
+                        wildcard_mention_notify=(user_id in send_request.wildcard_mention_user_ids),
+                    )
+                )
+
```

Comment:
We can clean this up to avoid duplicating code -- I think we just need to turn the set of `user_ids` into a list with the `sender_id` at the front.  E.g. something like this would work:
```
if sender_id in user_ids:
     user_list = [sender_id] + list(user_ids - {sender_id})
else:
     user_list = list(user_ids)
```

Reply:
Yes, It is definitely the cleaner way to do it.
Changed it

Reply:
You didn't do this as I'd written; it doesn't super matter, but I wanted to avoid creating the list twice, since it can be a list of 20K elements, and the fewer copies we do on those, the better.  Changed this and pushed back here.